### ISSUE TITLE
TensorMemoryLayout::None

### ISSUE THREAD

**Issue by jnie-TT** [tenstorrent/tt-mlir#1092]
Followup issue for #840, currently using this to model the TensorMemoryLayout of tensors on host. We should be able to remove this once we have the TTNN-specific tensor attr in our TTNN dialect, as we could just make the entire MemoryConfigAttr optional on the tensor.

Blocked by: #701

**Comment by jnie-TT** [comment]
Fixed by #1502
